### PR TITLE
Scodec upgrade

### PR DESCRIPTION
**Pull request by jedesah**
Updated to merge with current master

### CODE REVIEW

File: core/src/main/scala/codecs/codecs.scala [hunk 1]
```diff
@@ -51,72 +52,76 @@ package object codecs extends lowerprioritycodecs with TupleHelpers {
   implicit val utf8 = C.variableSizeBytes(int32, C.utf8)
   implicit val bool = C.bool(8) // use a full byte
 
-  implicit def tuple2[A:Codec,B: Codec]: Codec[(A,B)] =
-    Codec[A] ~ Codec[B]
+  implicit def tuple2[A, B](implicit LCA: Lazy[Codec[A]], LCB: Lazy[Codec[B]]): Lazy[Codec[(A,B)]] =
```
Comment (ahjohannessen) [inline core/src/main/scala/codecs/codecs.scala]: @jedesah if you upgrade to scodec 1.7.1 then I think return type of tuple2 does not need to be Lazy

Comment (ahjohannessen) [inline core/src/main/scala/codecs/codecs.scala]: @jedesah I tested it out, seems like to be a shapeless issue that most likely is solved in 2.2. So disregard my comment about changing to 1.7.1.

Comment (ahjohannessen) [inline core/src/main/scala/codecs/codecs.scala]: Great work btw :+1:

Comment (jedesah) [inline core/src/main/scala/codecs/codecs.scala]: Most of it is yours! :-)

File: core/src/main/scala/Server.scala [hunk 2]
```diff
@@ -57,8 +61,8 @@ object Server {
           val deltaNanos = System.nanoTime - startNanos
           val delta = Duration.fromNanos(deltaNanos)
           val result = right(a)
-          monitoring.handled(ctx, r, expected, result, delta)
-          toTask(codecs.responseEncoder(respEncoder).encode(result))
+          monitoring.handled(ctx, r, expected, right(a), delta)
```
Comment (ahjohannessen) [inline core/src/main/scala/Server.scala]: @jedesah I forgot to use result in handled, it would be great if you could update the code :-)

Comment (jedesah) [inline core/src/main/scala/Server.scala]: Sure, will do.

Comment (jedesah) [inline core/src/main/scala/Server.scala]: Done!

**Comment by ahjohannessen** [comment]
Oh Yeah :)

**Comment by ahjohannessen** [comment]
@stew @timperrett Could we get a 1.2.11 release published? :sunglasses:

**Comment by timperrett** [comment]
@ahjohannessen sure, i'll try to do it now.

**Pull request merged**